lua/autorun/server/ji_dsserver.lua

```diff
@@ -105,6 +105,19 @@ function JID.DetermineAttacker( ent )
     return ent
 end
 
+function JID.CanConstrain( ent, toConstrain )
+    if not CPPI then return true end
+
+    if not IsValid( ent ) then return end
+    if not IsValid( toConstrain ) then return end
+
+    local toConstrainOwner = toConstrain:CPPIGetOwner()
+    if not IsValid( toConstrainOwner ) then return end
+
+    local entOwner = ent:CPPIGetOwner()
+    if entOwner == toConstrainOwner then return true end
+end
```

Comment:
Proper usage of CPPI, check edge cases
```suggestion
function JID.CanConstrain( ent, toConstrain )
    if not IsValid( ent ) then return end
    if not IsValid( toConstrain ) then return end

    local entOwner = CPPI and ent:CPPIGetOwner() or ent:GetOwner()
    if not IsValid( entOwner ) then return end
    if CPPI and not toConstrain:CPPICanTool( entOwner ) then return end

    local toConstrainOwner = CPPI and toConstrain:CPPIGetOwner() or toConstrain:GetOwner()
    if not IsValid( toConstrainOwner ) then return end
    if CPPI and not ent:CPPICanTool( toConstrainOwner ) then return end

    if CPPI then return true end -- CPPI exists and hasn't blocked, so allow
    if entOwner == toConstrainOwner then return true end -- CPPI doesn't exist, directly compare owners
end
```

Reply:
can you add
```lua
local CPPI = CPPI 
```

Reply:
Done